tests/integration_tests.rs

```diff
@@ -0,0 +1,113 @@
+use k8s_openapi::api::core::v1::{Node, Pod};
+use kube::{
+    api::{Api, PostParams},
+    client::APIClient,
+    config,
+};
+use serde_json::json;
+
+#[tokio::test]
+async fn test_wascc_provider() {
+    // Read the environment. Note that this tries a KubeConfig file first, then
+    // falls back on an in-cluster configuration.
+    let kubeconfig = config::load_kube_config()
+        .await
+        .or_else(|_| config::incluster_config())
+        .expect("kubeconfig failed to load");
+
+    let client = APIClient::new(kubeconfig);
+
+    let nodes: Api<Node> = Api::all(client.clone());
+
+    let node = nodes
+        .get("krustlet-wascc")
+        .await
+        .expect("failed to find node with name 'krustlet-wascc'");
+
+    let node_status = node.status.expect("node reported no status");
+    assert_eq!(
+        node_status
+            .node_info
+            .expect("node reported no information")
+            .architecture,
+        "wasm-wasi",
+        "expected node to support the wasm-wasi architecture"
+    );
+
```

Comment:
There a reason we aren't creating a pod here?

Reply:
no particular reason